tests/functionnal/Log.php

```diff
@@ -717,4 +722,127 @@ protected function dataFiltersValuesToSqlCriteria() {
    public function testConvertFiltersValuesToSqlCriteria($filters_values, $expected_result) {
       $this->array(\Log::convertFiltersValuesToSqlCriteria($filters_values))->isIdenticalTo($expected_result);
    }
+
+   protected function testGetExportNameProvider() {
+      $date = date('Y_m_d', time());
+      $computer = new Computer();
+      $id = $computer->add([
+         'name'        => 'testGetExportName',
+         'entities_id' => getItemByTypeName('Entity', '_test_root_entity', true),
+      ]);
+
+      $this->integer($id)->isGreaterThan(0);
+      $this->boolean($computer->getFromDB($id))->isTrue();
+
+      return [
+         [
+            'item'     => $computer,
+            'locale'   => "en_GB",
+            'expected' => "Computer_{$id}_{$date}.csv"
+         ],
+         [
+            'item'     => $computer,
+            'locale'   => "fr_FR",
+            'expected' => "Ordinateur_{$id}_{$date}.csv"
+         ]
+      ];
+   }
+
+   /**
+    * @dataProvider testGetExportNameProvider
+    */
+   public function testGetExportName(
+      CommonDBTM $item,
+      string $locale,
+      string $expected
+   ) {
+      // Switch to locale
+      $_SESSION["glpilanguage"] = Session::loadLanguage($locale);
+
+      // Run test
+      $filename = \Log::getExportName($item);
+      $this->string($filename)->isEqualTo($expected);
+
+      // Reset locale
+      $_SESSION["glpilanguage"] = Session::loadLanguage('en_GB');
```

Comment:
```suggestion
      // Reset locale
      $_SESSION["glpilanguage"] = Session::loadLanguage('en_GB');
      
      $this->string($filename)->isEqualTo($expected);
```

Reply:
I don't get this one.

Reply:
You have to restet before test, then if the test fails, reset is already done, and next run will be ok.

Reply:
(when using atoum' `loop` flag)

Reply:
The tests is running with different locales (supplied by the provider function).

Process is:
- load locale
- test the string
- reset locale

Unless I'm missing something it's not possible to alter this order.

Reply:
```
$this->string($filename)->isEqualTo($expected);
// Reset locale
$_SESSION["glpilanguage"] = Session::loadLanguage('en_GB');
```

With that, if the test fails, locale is not reset because the code will never be reached; you will end with a language in session that is not the one excepted.